src/Common/AsynchronousMetrics.cpp

```diff
@@ -921,6 +921,15 @@ void AsynchronousMetrics::processWarningForMemoryOverload(const AsynchronousMetr
     double memory_resident{};
     double memory_total{};
 
+    /// Compute the user space page cache size (in bytes) and subtract from total memory used if possible
+    double user_space_cache_bytes{};
+    if (const auto *rss = getAsynchronousMetricValue(new_values, "MemoryResident"),
+        *rss_wo_pc = getAsynchronousMetricValue(new_values, "MemoryResidentWithoutPageCache");
```

Comment:
Looks OK

Another option is to add `CGroupMemoryUsedWithoutPageCache` and use it or `MemoryResidentWithoutPageCache` here